rfcs/quic.md

```diff
@@ -0,0 +1,104 @@
+# RFC 42: Add an optional QUIC server to wpt
+
+## Summary
+
+Add an **optional** [QUIC](https://quicwg.org/) server to wpt, implemented with
+[aioquic](https://github.com/aiortc/aioquic) in Python 3 and listening on port
+`9001` by default. This server is used to test web APIs that rely on the
+QUIC protocol (e.g. [WebTransport](https://wicg.github.io/web-transport/)). The
+server is optional but tries to start by default; `wpt` front-end will skip the
+tests that rely on QUIC if the QUIC server cannot be started (e.g. due to
+missing dependencies).
+
+QUIC custom handlers use a new filename flag `.quic` (`.quic.py`). And a meta
+tag is included by tests that require the QUIC server.
+
+See the [original issue][original-issue] for more background.
+
+## Details
+
+### Requirements
+
+1.  The QUIC server requires decent cryptography support; pure Python isn't
+    sufficient.
+2.  The server needs to be scriptable, and the custom handlers should live in
+    the wpt repo next to test files, so that they can be updated together.
+3.  An async programming model is preferred.
+
+### Architecture
+
+The QUIC server has a standalone CLI, and is managed by `wpt serve` as a
+subprocess. Configurations are stored in `config.json` (same as `wptserve`), and
+`wpt serve` is responsible for passing the relevant configurations (e.g. port
+numbers) to the QUIC server via command line flags. `wpt serve` handles the
+error gracefully if the QUIC server cannot be started.
+
+The QUIC server is written in Python 3 and makes no attempt to support Python 2.
+Its entrypoint will have a shebang that explicitly requires `python3`. An
+implication is that the QUIC server will be relatively self contained, without
+dependencies on other modules in wpt that are currently Python 2-only.
+
+Code for the QUIC server itself will live in `tools/quic`.
+
+### User interface
+
+For **test writers**: the QUIC server will support Python custom handlers
+similar to `wptserve`. The APIs will likely be different. To more easily
```

Comment:
I think we should have details of what the actual API will be as part of this or some other RFC.